keps/sig-network/2595-expanded-dns-config/README.md

```diff
@@ -0,0 +1,725 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Allow kubernetes to have expanded DNS(Domain Name System) configuration.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Kubernetes today limits DNS configuration according to the obsolete
+criteria(https://access.redhat.com/solutions/58028). As recent DNS resolvers
+allow an arbitrary number of search paths, a new feature gate
+`ExpandedDNSConfig` will be introduced. With this feature, kubernetes allows
+more DNS search paths and longer list of DNS search paths to keep up with recent
+DNS resolvers.
+
```

Comment:
maybe you should mention here that this is supported on (quoting) from https://github.com/kubernetes/kubernetes/issues/100622
```
 Confirm various DNS resolvers
 glibc 2.17-323
 glibc 2.28
 musl 1.22
 pure go 1.10 resolver
 pure go 1.16 resolver
 1.22: Prepare to relax the constraints of the PodDNSConfig - #100583
 1.23: Relax the constraints unconditionally
```

Reply:
Updated. Thanks!